lib/src/protocol/pipe.rs

```diff
@@ -196,28 +212,32 @@ impl<Front: SocketHandler, L: ListenerHandler> Pipe<Front, L> {
         }
     }
 
-    pub fn log_request(&self, metrics: &SessionMetrics, message: Option<&str>) {
+    pub fn log_request(&self, metrics: &SessionMetrics, error: bool, message: Option<&str>) {
```

Comment:
the `error` parameter should be a logging level to me, wdyt ?

Reply:
I think not. There are only two kinds of access logs we want to record here: info access, error access. The log level is applied down the line by the `log_access!` macro in the logging module:

```rust
#[macro_export]
/// dynamically chose between info_access and error_access
macro_rules! log_access {
    ($error:expr, $($request_record_fields:tt)*) => {
        let lvl = if $error {
            $crate::logging::LogLevel::Error
        } else {
            $crate::logging::LogLevel::Info
        };
        _log_access!(lvl, $($request_record_fields)*);
    };
}
```
It is wiser to leave this business in the log_access macro, in my opinion.